hash_tables/hash_table.py

```diff
@@ -0,0 +1,135 @@
+"""Create a Hash Table class with linked lists as buckets."""
+
+
+class HashNode:
+    """Create the node class for a hash_linked_list."""
+
+    def __init__(self, key, val, next=None):
+        """Identify this as a constructor a hash table Linked List node."""
+        if val is None:
+            raise TypeError('The value cannot be none')
+
+        self.val = val
+        self.key = key
+        self._next = next
+
+    def __str__(self):
+        """Make the str method for the HashNode."""
+        return str(self.val)
+
+    def __repr__(self):
+        """Make the repr method for the HashNode."""
+        return str(self.val)
+
+
+class HashLinkedList:
+    """Create the hash_linked_list class."""
+
+    def __init__(self):
+        """Identify this as a constructor a hash table Linked List node."""
+        self.head = None
+        self._size = 0
+
+    def __repr__(self):
+        """Return head and size of linked list when called."""
+        return '<head> => {}, <_size> => {}'.format(self.head.val, self._size)
+
+    def __len__(self):
+        """Return linked list length when called."""
+        return self._size
+
+    def insert(self, key, val):
+        """Insert a new value into the bucket at ther head of the list."""
+        try:
+            node = HashNode(key, val)
+        except TypeError:
+            raise TypeError('Cannot insert a value of none')
+
+        node._next = self.head
+        self.head = node
+        self._size += 1
+
+    def remove(self, key):
+        """Remove the node from the linked_list that matches the key and return the node."""
+        current = self.head
+        while current.key != key:
```

Comment:
This assumes that the given key is in the list. If the key is not in the list, this will end up raising and error.